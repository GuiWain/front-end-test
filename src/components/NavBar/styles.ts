import styled, { css } from 'styled-components'
import { media } from 'styles/media'

import UseLink from 'next/link'

export const Link = styled(UseLink)``

export const Nav = styled.nav`
  width: 100%;

  ${media.lessThan('desktop')`
      height: 100%;
  `}
`

export const List = styled.ul`
  ${() => css`
    list-style: none;
    display: flex;
    justify-content: flex-end;
    align-items: center;
    margin: 0 auto;

    ${media.lessThan('desktop')`
      height: 100%;
      flex-direction: column;
      justify-content: space-around;
    `}
  `}
`

export const Item = styled.li`
  ${({ theme }) => css`
    font-weight: ${theme.font.bold};
    font-size: ${theme.font.sizes.xxsmall};
    color: ${theme.colors.secondaryMedium};
    text-align: center;
    margin: 0 1rem;
    cursor: pointer;
    transition: color ${theme.transition.default};

    &:hover {
      color: ${theme.colors.secondaryDark};
      transition: color ${theme.transition.default};
    }
  `}
`
