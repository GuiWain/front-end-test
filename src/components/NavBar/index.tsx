import * as S from './styles'

const NavBar = ({ ...props }) => (
  <S.Nav {...props}>
    <S.List>
      <S.Link href="#">
        <S.Item>Link 1</S.Item>
      </S.Link>
      <S.Link href="#">
        <S.Item>Link 2</S.Item>
      </S.Link>
      <S.Link href="#">
        <S.Item>Link 3</S.Item>
      </S.Link>
      <S.Link href="#">
        <S.Item>Link 4</S.Item>
      </S.Link>
      <S.Link href="#">
        <S.Item>Link 5</S.Item>
      </S.Link>
      <S.Link href="#">
        <S.Item>Link 6</S.Item>
      </S.Link>
    </S.List>
  </S.Nav>
)

export default NavBar
