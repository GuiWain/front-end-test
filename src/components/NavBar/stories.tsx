import { Story, Meta } from '@storybook/react'
import NavBar from '.'

export default {
  title: 'Components/NavBar',
  component: NavBar
} as Meta

export const Default: Story = () => (
  <div style={{ height: '70vh', margin: 'auto' }}>
    <NavBar />
  </div>
)
