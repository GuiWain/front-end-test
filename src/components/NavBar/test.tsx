import { render, screen } from 'utils/test-utils'

import NavBar from '.'

describe('<NavBar />', () => {
  it('should render the NavBar', () => {
    const { container } = render(<NavBar />)

    expect(screen.getAllByText(/Link/i)).toHaveLength(6)
    expect(container.firstChild).toMatchSnapshot()
  })
})
