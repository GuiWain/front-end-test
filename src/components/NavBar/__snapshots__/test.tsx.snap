// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> should render the NavBar 1`] = `
.c0 {
  width: 100%;
}

.c1 {
  list-style: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0 auto;
}

.c2 {
  font-weight: 700;
  font-size: 1.4rem;
  color: #848080;
  text-align: center;
  margin: 0 1rem;
  cursor: pointer;
  -webkit-transition: color 0.5s ease-in-out;
  transition: color 0.5s ease-in-out;
}

.c2:hover {
  color: #4F4C4C;
  -webkit-transition: color 0.5s ease-in-out;
  transition: color 0.5s ease-in-out;
}

@media (max-width:1024px) {
  .c0 {
    height: 100%;
  }
}

@media (max-width:1024px) {
  .c1 {
    height: 100%;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-pack: space-around;
    -webkit-justify-content: space-around;
    -ms-flex-pack: space-around;
    justify-content: space-around;
  }
}

<nav
  class="c0"
>
  <ul
    class="c1"
  >
    <li
      class="c2"
    >
      Link 1
    </li>
    <li
      class="c2"
    >
      Link 2
    </li>
    <li
      class="c2"
    >
      Link 3
    </li>
    <li
      class="c2"
    >
      Link 4
    </li>
    <li
      class="c2"
    >
      Link 5
    </li>
    <li
      class="c2"
    >
      Link 6
    </li>
  </ul>
</nav>
`;
